### ISSUE TITLE
Possible issue with submodules and/or the update script

### ISSUE THREAD

**Issue by brucehoward-physics** [gundam-organization/gundam#429]
Hi! I recently had to set GUNDAM back up in a new working area and was trying to get it back into a working order and possibly the same release that I'd used before. (This is not really a requirement, but is what I tried.) When I did this I was not able to build (compiler throws errors in the GUNDAM bits), and it seems traceable to the submodules and/or the interaction with the updates.sh script.

I believe when I set up the first time I will have been using
`./setup.sh --latest`
But since my working area is telling me I have version 1.7.2 I have updated my installation shell script to use
`./update.sh -v 1.7.2`
When I do this, my build fails (I can post error message if useful, but I think/hope the below is pointing in the direction)

However, in the submodules I have very recent commits getting into my build (see below):

`cpp-generic-toolbox` :

```
commit 209f8a4bb37648c2fbc30d90130d4d6fd6a77a43 (HEAD, origin/main, origin/HEAD, main)
Author: Adrien Blanchet <adrien.blanchet@live.fr>
Date:   Mon Dec 4 17:13:09 2023 +0100

    more general definition
```

`simple-cpp-cmd-line-parser` is giving me something from May so that seems alright. Similarly `simple-cpp-logger` is giving me something from June so that is also probably ok. (Or these submodules haven't had recent commits, I didn't check to be fair.)

I was able to successfully build if I tell it to check out the specific commits that I was using in my initial setup. I.e. the installation script I wrote now does:
```
./update.sh -v 1.7.2
cd submodules/cpp-generic-toolbox/
git checkout f22c1a5
cd ../simple-cpp-cmd-line-parser
git checkout 8ac9331
cd ../simple-cpp-logger
git checkout 2a924a0
```

Thanks in advance for any info/help!

**Comment by riccioc** [comment]
Hi Bruce, thanks for using GUNDAM and opening this issue! Just to understand, you moved from the latest (the main branch) version to the tag 1.7.2 and the submodules didn't get sync with the tagged version?

**Comment by brucehoward-physics** [comment]
Hi Ciro!

Seemingly, or at least `cpp-generic-toolbox` specifically didn't.

Working example with the GUNDAM bits after I've done some other pieces:
```
git clone https://github.com/gundam-organization/gundam.git
cd gundam/
./update.sh -v 1.7.2
```

In this case I found that I get `1.7.2` of GUNDAM, but `cpp-generic-toolbox` specifically gets a very recent commit.

**Comment by nadrino** [comment]
Hi Bruce! Nice to meet you!

Those submodules are complicated to handle with the different git versions...

We're about to tag a new release of GUNDAM so I guess the issue should be solved.

If you want to stick to the 1.7.2, you could do it manually:
```bash
git clone https://github.com/gundam-organization/gundam.git
cd gundam/
git checkout 1.7.2
git submodule update --init --remote
```

**Comment by nadrino** [comment]
This should be fixed in the 1.9.0
